### ISSUE TITLE
Training class indices and order of model output

### ISSUE THREAD

**Issue by drnwtt** [malllabiisc/EWISE#8]
Hey, I have a question trying to understand the code. As far as I understand, the sense embedding matrix is ordered by the synset ids (see [conve/definition_processor.py](https://github.com/malllabiisc/EWISE/blob/master/conve/definition_preprocessor.py)) whereas the output tokens are indexed by their order of appearance in the corpus (see [index_file_creator.py](https://github.com/malllabiisc/EWISE/blob/master/index_file_creator.py)). When the model output (the distribution over the sense inventory) is compared to the desired output classes (`y`) during training
`loss = criterion(output_masked, y_masked)`
how is the correspondance achieved between the indices of annotated tokens in `y` and the output of the model. Are both assumed to have the same order, or is a mapping happening at some place?
Many thanks in advance!

**Comment by SawanKumar28** [comment]
[generate_output_embeddings.py](https://github.com/malllabiisc/EWISE/blob/master/generate_output_embeddings.py) converts the output from the definition processor to the same space as [index_file_creator.py](https://github.com/malllabiisc/EWISE/blob/master/index_file_creator.py) (See step 3 in the README).
